### DISCUSSION TITLE
CMake can't find Libmad although already installed?

### DISCUSSION THREAD

**Discussion by iijj22gg** [stepmania/stepmania#2157] [Q&A]
Hello!

I'm trying to compile Stepmania 5.1 on Linux (with MP3 support), and CMake is telling me it can't find Libmad.

I have installed the relevant package, updated the initramfs and rebooted, yet I'm getting the same error.

Is there something I'm missing?

**Comment by namgo** [comment]
Hi, can you confirm that you have the library's development package installed? I attempted to reproduce this issue with a clean build in Fedora, libmad-devel was required in addition to libmad package.

**Comment by user340q** [comment]
I am running Ubuntu 22.04.2 LTS.  I found I could not compile Stepmania 5.1.0 Beta 2 without external steps. This page was very helpful: https://heavydeck.net/blog/building-an-stepmania-appimage/

1. The list of packages to install via apt was different on that page than the Stepmania instructions, and installed a few extra things not found when copy pasting the commands from the Stepmania wiki
``~/
$ apt-get install \
	libmad0-dev libvorbis-dev libbz2-dev \
	libx11-dev zlib1g-dev libjpeg62-dev \
	libxinerama-dev libasound-dev libpulse-dev \
	libva-dev libglew-dev libxrandr-dev \
	libudev-dev libxtst-dev yasm``
2. You must edit SetupFfmpeg.cmake in the Cmake folder - on line 72, you must change the url from "**git:/**/github.com/stepmania/ffmpeg.git" to "**https:/**/github.com/stepmania/ffmpeg.git" (this appeared to be fixed on 5_1-new, but not 5.1.0b2) or it will never finish compiling.
3. The cmake command I successfully used is ``cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr ..``
4. I then ran ``make -j5`` and everything worked great.

Hope that helps.  (Edited to fix a markup error)
